keps/sig-auth/3257-cluster-trust-bundles/README.md

```diff
@@ -563,6 +579,64 @@ For example, if I maintain `example.com/my-signer`, I can use the following stra
   canary object first, and assess the health of the canary workloads.
 * Once I am satisfied that the change is safe, I edit the live object.
 
+### Publishing the kube-apiserver-serving Trust Bundle
+
+Today, the trust bundle that allows verifying kube-apiserver serving certificate(s)
+at its internal endpoints is distributed into every namespace using a configMap.
+This is so that it can be mounted along with the ServiceAccount token in order
+for the workloads to be able to communicate with the kube-apiserver.
+
+In the future, we should be able to replace mounting these configMaps in pods for
+for kube-apiserver trust with the projected volume from this feature, and so a
+ClusterTrustBundle API object will be created for all clusters:
+
+```yaml
+apiVersion: v1beta1
+kind: ClusterTrustBundle
+metadata:
+  generateName: kubernetes.io:kube-apiserver-serving:
+spec:
+  signerName: kubernetes.io/kube-apiserver-serving
+  trustBundle: "<... PEM CA ...>"
+```
+
+This object is managed by the Kubernetes Controller Manager in the existing
+`root-ca-certificate-publisher-controller`. It serves the same purpose
+and contains the same content as the `ca.crt` data in the `kube-root-ca.crt` configMap - to verify internal kube-apiserver
+endpoints. There is currently no in-tree signer designated for these purposes,
+and so the signer with name `kubernetes.io/kube-apiserver-serving` is introduced
+along with this bundle.
+
+This behavior is feature-gated by the `ClusterTrustBundle` KCM feature gate.
+
+#### The kube-apiserver-serving signer
+
+The signer signs certificates that can be used to verify kube-apiserver serving
+certificates. Signing and approval are handled outside kube-controller-manager.
+
+**Trust distribution** - signed certificates are used by the kube-apiserver for TLS
+server authentication. The CA bundle is distributed using a ClusterTrustBundle object
+identifiable by the `kubernetes.io/kube-apiserver-serving` signer name.
+**Permitted subjects** - "Subject" itself is deprecated for TLS server authentication. However,
+it should still follow the same rules on DNS/IP SANs from the "Permitted x509 extensions" section
+below.
+**Permitted x509 extensions** - honors subjectAltName and key usage extensions. At
+least one DNS or IP subjectAltName must be present. The SAN DNS/IP of the certificates
+must resolve/point to kube-apiserver's hostname/IP.
+**Permitted key usages** - ["key encipherment", "digital signature", "server auth"] or ["digital signature", "server auth"].
+**Expiration/certificate lifetime** - The recommended maximum lifetime is 30 days.
```

Comment:
Even if you ignore kubeconfigs distributed to end users, https://github.com/kubernetes/kubernetes/issues/119483 makes it hard to rotate this CA for in-cluster workloads.

Reply:
IIUC the section describes the certs signed by the certs of the bundle.